charts/sourcegraph/values.yaml

```diff
@@ -221,6 +221,15 @@ frontend:
     create: true
     name: sourcegraph-frontend
 
+migrator:
```

Comment:
I wonder if we should also enable running the migrator as the standalone Job (not associated with helm hooks, just as an option). This would be in addition to the initContainer, and accept a custom arg.

The reason for this would be to make it possible to use the chart to fix database migration errors (https://docs.sourcegraph.com/admin/how-to/dirty_database#3-add-a-migration-log-entry), or to allow customers to deploy the migration job ahead of a general upgrade.

Reply:
how about

## Option 1
```yml
migrator:
  type: "" # `initContainer` or `job`
```

Then we render `initContainers` or a new `Job` depending on the values

## Option 2

Create a new chart `sourcegraph/migrator`, by doing so, it makes people's life easier to just run the migration. This could be quite helpful for stuff like manually fixing a dirty migration

```bash
helm upgrade sourcegraph-migrator sourcegraph/migrator --values args=<>
```

or

just render the manifest if the client doesn't want to use helm to update stuff

```bash
helm template sourcegraph-migrator sourcegraph/migrator --values args=<>
```

Reply:
Option 1 doesn't quite work - we need to be able to run both `initContainer` and `Job` at the same time, and with different args.

Option 2 is interesting. As a standalone chart it's easily uninstalled after the migration runs. It could even be called as an optional dependency of this chart, so that env var's could be reused instead of having to specify them separately. Let's experiment with this approach and see if any surprises come up!